perf/src/bin/perf_server.rs

```diff
@@ -100,18 +106,33 @@ async fn run(opt: Opt) -> Result<()> {
     Ok(())
 }
 
-async fn handle(handshake: quinn::Connecting) -> Result<()> {
+async fn handle(handshake: quinn::Connecting, opt: Arc<Opt>) -> Result<()> {
     let quinn::NewConnection {
         uni_streams,
         bi_streams,
         connection,
         ..
     } = handshake.await.context("handshake failed")?;
     debug!("{} connected", connection.remote_address());
-    tokio::try_join!(drive_uni(connection, uni_streams), drive_bi(bi_streams))?;
+    tokio::try_join!(
+        drive_uni(connection.clone(), uni_streams),
+        drive_bi(bi_streams),
+        conn_stats(connection, opt)
+    )?;
     Ok(())
 }
 
+async fn conn_stats(connection: quinn::Connection, opt: Arc<Opt>) -> Result<()> {
+    loop {
+        tokio::time::sleep(Duration::from_secs(2)).await;
+        {
+            if opt.conn_stats {
+                println!("{:?}\n", connection.stats());
+            }
+        }
+    }
```

Comment:
Could we move this branch outside of the loop?

Reply:
Sure. I first thought the function may not end in order not to disconnect the server prematurely - but it's ok since it uses `try_join!` and not `select!`